lib/promrelabel/scrape_url.go

```diff
@@ -38,12 +39,25 @@ func GetScrapeURL(labels *promutils.Labels, extraParams map[string][]string) (st
 		address = address[:n]
 	}
 
+	// Always escape IPv6 addresses. It is safe to always escape IPv6 addresses
+	// and this makes further usage of the address easier.
+	isIPv6Address := isIPv6(address)
```

Comment:
It looks like the real issue is at Kubernetes discovery.  If port is not defined for pod, vmagent uses IP address without formatting. And it breaks scrape url builder.

 In any other cases it joins port and properly escapes ipv6 addresses:
https://github.com/VictoriaMetrics/VictoriaMetrics/blob/b27e9437a3ba6bd8a3d9ae7d2f1e7f091625be5f/lib/promscrape/discovery/kubernetes/pod.go#L198C25-L198C37

https://github.com/VictoriaMetrics/VictoriaMetrics/blob/b27e9437a3ba6bd8a3d9ae7d2f1e7f091625be5f/lib/promscrape/discoveryutils/utils.go#L31

Reply:
I think proposed change could be a nice feature. It should simplify relabeling of ipv6 address.

Reply:
> It looks like the real issue is at Kubernetes discovery. If port is not defined for pod, vmagent uses IP address without formatting. And it breaks scrape url builder.

It seems to me like scrape URL builder should handle both IPv4 and IPv4 addresses as the same can happen with other discovery mechanisms later on.

I've also simplified IPv6 address check based on how discoveryutils handle it.

Reply:
If some service discovery mechanism doesn't perform wrap [] for ipv6 address label, I consider it as a bug.

A simple example, could you please write relabeling config for both wrapped and not-wrapped ipv6 addresses? 

 More over, `fd01:10:100:0:e38f:6f4b:9c53:9e4e:PORT_NUMBER` is invalid syntax and we cannot do anything with it.

 Btw, the last fix is incorrect and it fails for simple case.

Reply:
Moved a fix to `discovery/kubernetes` side. PTAL